lib/transformers.mli

```diff
@@ -0,0 +1,51 @@
+open Any
+open Control
+
+type ('r, 'am) readerT
+(** The Reader and ReaderT monads.
+    For ReaderT, transforming an inner monad m, use ('r, 'a m) readerT
+    For Reader, with no inner monad, use ('r 'a) readerT
+ *)
```

Comment:
This is making me think that `reader` and `readerT` should be separate types? In fact it was kind of an accident that I used `readerT` for the non-transformed version too.